packages/push-platform-notifications-controller/src/utils/get-notification-message.ts

```diff
@@ -0,0 +1,260 @@
+      });
+      return `You successfully sent ${amount} ${symbol}`;
+    },
+  },
+  erc20_received: {
+    title: 'Funds received',
+    defaultDescription: 'You received some tokens',
+    getDescription: (n) => {
+      const symbol = n?.data?.token?.symbol;
+      const tokenAmount = n?.data?.token?.amount;
+      const tokenDecimals = n?.data?.token?.decimals;
+      if (!symbol || !tokenAmount || !tokenDecimals) {
+        return null;
+      }
+
+      const amount = getAmount(tokenAmount, tokenDecimals, {
+        shouldEllipse: true,
+      });
+      return `You received ${amount} ${symbol}`;
+    },
+  },
+  eth_received: {
+    title: 'Funds received',
+    defaultDescription: 'You received some tokens',
+    getDescription: (n) => {
+      const symbol = getChainSymbol(n?.chain_id);
+      const tokenAmount = n?.data?.amount?.eth;
+      if (!symbol || !tokenAmount) {
+        return null;
+      }
+
+      const amount = formatAmount(parseFloat(tokenAmount), {
+        shouldEllipse: true,
+      });
+      return `You received ${amount} ${symbol}`;
+    },
+  },
+  metamask_swap_completed: {
+    title: 'Swap completed',
+    defaultDescription: 'Your MetaMask Swap was successful',
+  },
+  erc721_sent: {
+    title: 'NFT sent',
+    defaultDescription: 'You have successfully sent an NFT',
+  },
+  erc1155_sent: {
+    title: 'NFT sent',
+    defaultDescription: 'You have successfully sent an NFT',
+  },
+  erc721_received: {
+    title: 'NFT received',
+    defaultDescription: 'You received new NFTs',
+  },
+  erc1155_received: {
+    title: 'NFT received',
+    defaultDescription: 'You received new NFTs',
+  },
+  rocketpool_stake_completed: {
+    title: 'Stake complete',
+    defaultDescription: 'Your RocketPool stake was successful',
+  },
+  rocketpool_unstake_completed: {
+    title: 'Unstake complete',
+    defaultDescription: 'Your RocketPool unstake was successful',
+  },
+  lido_stake_completed: {
+    title: 'Stake complete',
+    defaultDescription: 'Your Lido stake was successful',
+  },
+  lido_stake_ready_to_be_withdrawn: {
+    title: 'Stake ready for withdrawal',
+    defaultDescription: 'Your Lido stake is now ready to be withdrawn',
+  },
+  lido_withdrawal_requested: {
+    title: 'Withdrawal requested',
+    defaultDescription: 'Your Lido withdrawal request was submitted',
+  },
+  lido_withdrawal_completed: {
+    title: 'Withdrawal completed',
+    defaultDescription: 'Your Lido withdrawal was successful',
+  },
+};
+
+/**
+ * Creates a push notification message based on the given on-chain raw notification.
+ *
+ * @param n - The on-chain raw notification object.
+ * @returns The push notification message object, or null if the notification is invalid.
+ */
+export function createNotificationMessage(
+  n: OnChainRawNotification,
+): PushNotificationMessage | null {
+  if (!n?.data?.kind) {
+    return null;
+  }
+  const notificationMessage = notificationMessageDict[n.data.kind] as
```

Comment:
Why do we need a typecast here and is there a way to avoid it?

Reply:
We don't. Removed.